### DISCUSSION TITLE
Ask about camera ssk parameter

### DISCUSSION THREAD

**Discussion by ThanhTrung128** [UTokyo-FieldPhenomics-Lab/EasyIDP#99] [Q&A]
Hi author!
I have a question about camera ssk. Can you help me how to calculate some parameter of camera ssk?
Currently, I have some input as dom file (tiff), dsm file (tiff), point cloud (.ply) and image folder. I want to create Backward Projection ROI from point cloud file to original images. But I don't know how to create param folder as Lotus Pix4D dataset Because I don't have more knowledge about photo, or camera parameter. Can you help me?

**Comment by HowcanoeWang** [comment]
Hi, 

The Lotus pix4d file structure may mislead you, I guess you are asking about the `param_folder=lotus.pix4d.param` right?

```python
p4d = idp.Pix4D(project_path=lotus.pix4d.project,
                raw_img_folder=lotus.photo,
                param_folder=lotus.pix4d.param)
```

If you check `lotus.pix4d.project`, `lotus_photo`, `lotus.pix4d.param` using `print()` command, they are just path strings. And the `param_folder` is just optional, to tell the easyidp where is the pix4d project parameter folder

E.g.

```plaintxt
/Users/..../xxxx.p4d  -> this is the pix4d project file
/Users/..../xxxx/     -> this is the pix4d parameter folder, by default no need to specify it
```

The Lotus Dataset gives that parameter because sometimes it needs to rearrange the folder structure for the data archive.

---

What you need to prepare later, is the `*.shp` file to represent the region that you want to convert. Since you already have the DOM and DSM file, it is recommended to use QGIS or ArcGIS to great that region, much easier than on pointcloud!

Please refer to this tutorial about how to make `*.shp` file:

https://easyidp.readthedocs.io/en/latest/backgrounds/roi_marking.html#marking-roi-on-gis-maps

**Comment by luongphambao** [comment]
Is it possible to use the backward projection (back2raw) method for custom dataset if I don't have access to the same parameters as those in **'lotus.pix4d.param'**? Specifically, I have a shapefile, point cloud, DSM, and DTM that I would like to use.

**Reply by HowcanoeWang** [reply]
as mentioned above, what you need to do, should be:

```python
p4d = idp.Pix4D(project_path=r"path/to/your/pix4d_project.p4d",
                raw_img_folder=r"path/to/your/uav_image_folder",
                param_folder=r"path/to/your/pix4d_project/")
```

**Reply by ThanhTrung128** [reply]
Can you share some knowledge how to create some parameter file in param folder?

**Reply by HowcanoeWang** [reply]
it is a folder created automatically by pix4d software when you save the project.

**Reply by HowcanoeWang** [reply]
To be more specific, it locates in `pix4d_project/1_initial/params`

Either 

```python
p4d = idp.Pix4D(project_path=r"path/to/your/pix4d_project.p4d",
                raw_img_folder=r"path/to/your/uav_image_folder")
# no need to handle that path by default
```

or 

```python
p4d = idp.Pix4D(project_path  =r"path/to/your/pix4d_project.p4d",
                raw_img_folder=r"path/to/your/uav_image_folder",
                param_folder  =r"path/to/your/pix4d_project/1_initial/params")
```

should work
